models/election.rb

```diff
@@ -1,15 +1,86 @@
 class Election < ActiveRecord::Base
-  has_many :calculations, as: :subject
+  include HasCalculations
 
-  def calculation(name)
-    @_calculations_cache ||= calculations.index_by(&:name)
-    @_calculations_cache[name.to_s].try(:value)
+  has_many :candidates, foreign_key: :election_name, primary_key: :name
+  has_many :office_elections, foreign_key: :election_name, primary_key: :name
+  has_many :referendums, foreign_key: :election_name, primary_key: :name
+
+  def locality
+    name.split('-', 2).first
+  end
+
+  def metadata_path
+    "/_elections/#{locality}/#{date}.md"
+  end
+
+  def metadata
+    {
+      'title' => title,
+      'election_id' => name,
+      'locality' => locality,
+      'election' => date.to_s,
+      'office_elections' => office_elections.group_by(&:label).map do |label, office_elections|
+        {
+          'label' => label,
+          'items' => office_elections.map do |office_election|
+            "_office_elections/#{locality}/#{date}/#{slugify(office_election.title)}.md"
+          end
+        }.compact
+      end,
+      'referendums' => referendums.map do |referendum|
+        "_referendums/#{locality}/#{date}/#{slugify(referendum.Short_Title)}.md"
+      end,
+    }
   end
 
-  def save_calculation(name, value)
-    calculations
-      .where(name: name)
-      .first_or_create
-      .update_attributes(value: value)
+  def data
+    {
+      'total_contributions_by_source' => {
+        'From Within Oakland' => 0.55,
```

Comment:
I don't get this.  Other places the def data segment looks reasonable, what are these constants for?

Reply:
Ah, those came over from this PR: https://github.com/caciviclab/disclosure-backend-static/pull/235

It's fake data so the frontend can start building the UI while we work on building the actual calculations.